src/lib.rs

```diff
@@ -2582,16 +2582,44 @@ impl Build {
                 "Detecting {:?} SDK path for {}",
                 os, sdk_details.sdk
             ));
-            let sdk_path = if let Some(sdkroot) = env::var_os("SDKROOT") {
-                sdkroot
-            } else {
-                self.apple_sdk_root(&sdk_details.sdk)?
-            };
+            let sdk_path = self.apple_sdk_root(&sdk_details.sdk)?;
 
             cmd.args.push("-isysroot".into());
             cmd.args.push(sdk_path);
         }
 
+        if let AppleArchSpec::Catalyst(_) = arch {
+            // Mac Catalyst uses the macOS SDK, but to compile against and
+            // link to iOS-specific frameworks, we should have the support
+            // library stubs in the include and library search path.
+            let sdk_path = self.apple_sdk_root(&sdk_details.sdk)?;
+            let ios_support = PathBuf::from(sdk_path).join("/System/iOSSupport");
+
+            cmd.args.extend([
+                // Header search path
+                OsString::from("-isystem"),
+                ios_support.join("/usr/include").into(),
+                // Framework header search path
+                OsString::from("-iframework"),
+                ios_support.join("/System/Library/Frameworks").into(),
+                // Library search path
+                {
+                    let mut s = OsString::from("-L");
+                    s.push(&ios_support.join("/usr/lib"));
+                    s
+                },
+                // Framework linker search path
```

Comment:
Should this get moved up to right under `-iframework`?

Reply:
Sorry didn't see this😂

Reply:
It doesn't really matter, I went with keeping the header and the library search paths separated.